flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/join/NonWindowJoin.scala

```diff
@@ -291,4 +253,9 @@ abstract class NonWindowJoin(
       joinFunction.join(otherSideRow, inputRow, cRowWrapper)
     }
   }
+
+  protected def cleanupState(states: State*): Unit = {
+    // clear all state
+    states.foreach(_.clear())
```

Comment:
do we need clean up the `cleanupTimeState` as well ?

Reply:
Do you mean clean up timer state here? This is an util function and will be called in other places.